src/widgets/FormInputs/FormDateInput.js

```diff
@@ -76,12 +76,9 @@ export const FormDateInput = React.forwardRef(
       const newValidity = onValidate(newValue);
       const updatedIsValid = validity && newValidity;
 
-      let updatedDate = null;
-      if (!newValue) {
-        updatedDate = null;
-      } else if (updatedIsValid) {
-        updatedDate = newDate.toDate();
-      }
+      const updatedDate = newValue && updatedIsValid ? newDate.toDate() : null;
```

Comment:
the date validation wasn't working, in that you could enter an invalid value, the field warns you, but the form `canSave` does not get set, as a `null` value is handed back to the form, which the form interprets as valid

Reply:
👍 

OK if you say so 🤣 , this component confuses me 😡